### ISSUE TITLE
TableView: keyboard traps with certain focusable components

### ISSUE THREAD

**Issue by jnurthen** [adobe/react-spectrum#4126]
# 🐛 Bug Report 

When including focusable objects in TableView there are focus traps for the following object types
* DateField
* DatePicker
* TimeField
* NumberField (not a trap but multiple keypresses appear to do nothing)
* RadioGroup (selection gets set unexpectedly)
* RangeSlider
* SearchField
* Slider
* Picker (not a trap but unexpected behaviour)
* Combobox (not a trap but unexpected behaviour)

## 🤔 Expected Behavior

There should be no keyboard traps 

## 😯 Current Behavior

* DateField - left/right arrow keys do not exit the component - we get stuck
* DatePicker - left/right arrow keys do not exit the component - we get stuck
* TimeField - - left/right arrow keys do not exit the component - we get stuck
* NumberField - It takes 3 left/right keypresses to exit the component and move to the next/previous with no indication that anything happens after each keypress
* RadioGroup - when using left/right to move through the component selection gets left on an unexpected entry
* RangeSlider - get trapped on the slider. Left/right move the slider. Not sure how this could ever work in a table as you can't even use the component on its own.
* SearchField - left/right arrow keys do not exit the component - we get stuck (Textfield works so this should probably be the same)
* Picker - focus moves but also changes the selection
* Combobox - left/right arrow keys do not exit the component - we get stuck 

## 💁 Possible Solution

Some just need fixing to allow focus to break out
Some others (range slider for example) should be forbidden in a TableView

## 💻 Code Sample

https://codesandbox.io/s/tableview-and-focusables-bed1t7

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3
| Browser          | Chrome
| Operating System | MacOS

**Comment by devongovett** [comment]
Components that use the arrow keys for something are not currently supported in TableView. We need to implement edit mode for them to work. #2328

**Comment by jnurthen** [comment]
> Components that use the arrow keys for something are not currently supported in TableView. We need to implement edit mode for them to work. #2328

This doesn't appear to be documented.

**Comment by LFDanLu** [comment]
Closing in favor of #2328
